lib/web/servers.go

```diff
@@ -335,3 +336,84 @@ func getDatabaseUsersAndNames(accessChecker services.AccessChecker) (dbNames []s
 type desktopIsActive struct {
 	Active bool `json:"active"`
 }
+
+// createNodeRequest contains the required information to create a Node.
+type createNodeRequest struct {
+	Name     string         `json:"name,omitempty"`
+	SubKind  string         `json:"subKind,omitempty"`
+	Hostname string         `json:"hostname,omitempty"`
+	Addr     string         `json:"addr,omitempty"`
+	Labels   []ui.Label     `json:"labels,omitempty"`
+	AWSInfo  *types.AWSInfo `json:"aws,omitempty"`
+}
+
+func (r *createNodeRequest) checkAndSetDefaults() error {
+	if r.Name == "" {
+		return trace.BadParameter("missing node name")
+	}
+
+	// Nodes provided by the Teleport Agent are not meant to be created by the user.
+	// They connect to the cluster and heartbeat their information.
+	//
+	// Agentless Nodes with Teleport CA call the Teleport Proxy and upsert themselves,
+	// so they are also not meant to be added from web api.
+	if r.SubKind != types.SubKindOpenSSHEC2InstanceConnectEndpointNode {
```

Comment:
`SubKindOpenSSHEC2InstanceConnectEndpointNode` lol this is a mouthful, how about renaming this to `SubKindEIC` (I know it's not a part of this PR but still)?

Reply:
What about `SubKindOpenSSHEIC`?
I think keeping the OpenSSH part is useful to give context, the rest is just a product/marketing name

Reply:
`SubKindOpenSSHEICE` sounds good to me.

Reply:
I'll go with `SubKindOpenSSHEICE`
Has an extra `E`, but matches the actual feature name: EC2 Instance Connect Endpoint

Let me know if you prefer the other option and I'll change it in the other PR